eclipse.jdt.ls build fails

See https://ci.eclipse.org/ls/job/jdt-ls-master/807/console

Hi, I encountered the same issue and It can be fixed by deleting "@Override" annotation of the method of

`public CompletableFuture<SignatureHelp> signatureHelp(TextDocumentPositionParams position) {}`

from 

org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java


my error message

[INFO] ------------------------------------------------------------------------

[INFO] Reactor Summary for JDT Language Server :: Parent 0.50.0-SNAPSHOT:

[INFO]

[INFO] JDT Language Server :: Parent ...................... SUCCESS [  0.306 s]

[INFO] JDT Language Server :: Target Platform ............. SUCCESS [  1.014 s]

[INFO] JDT Language Server :: Core ........................ FAILURE [  5.033 s]

[INFO] JDT Language Server :: Tests ....................... SKIPPED

[INFO] JDT Language Server :: Product ..................... SKIPPED

[INFO] ------------------------------------------------------------------------

[INFO] BUILD FAILURE

[INFO] ------------------------------------------------------------------------

[INFO] Total time:  42.869 s

[INFO] Finished at: 2020-01-31T20:01:00+09:00

[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.6.0:compile (default-compile) on project org.eclipse.jdt.ls.core: Compilation failure: Compilation failure:

[ERROR] /home/XXX/XXX/eclipse-workspace/eclipse.jdt.ls/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java:[534]

[ERROR]     public CompletableFuture signatureHelp(TextDocumentPositionParams position) {

[ERROR]                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[ERROR] The method signatureHelp(TextDocumentPositionParams) of type JDTLanguageServer must override or implement a supertype method

[ERROR] 1 problem (1 error)

[ERROR] -> [Help 1]

[ERROR]

[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.

[ERROR] Re-run Maven using the -X switch to enable full debug logging.

[ERROR]

[ERROR] For more information about the errors and possible solutions, please read the following articles:

[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

[ERROR]

[ERROR] After correcting the problems, you can resume the build with the command

[ERROR]   mvn  -rf :org.eclipse.jdt.ls.core

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java
@@ -124,6 +124,7 @@ import org.eclipse.lsp4j.RenameParams;
 import org.eclipse.lsp4j.SelectionRange;
 import org.eclipse.lsp4j.SelectionRangeParams;
 import org.eclipse.lsp4j.SignatureHelp;
+import org.eclipse.lsp4j.SignatureHelpParams;
 import org.eclipse.lsp4j.SymbolInformation;
 import org.eclipse.lsp4j.TextDocumentIdentifier;
 import org.eclipse.lsp4j.TextDocumentPositionParams;
@@ -531,7 +532,7 @@ public class JDTLanguageServer implements LanguageServer, TextDocumentService, W
 	 * @see org.eclipse.lsp4j.services.TextDocumentService#signatureHelp(org.eclipse.lsp4j.TextDocumentPositionParams)
 	 */
 	@Override
-	public CompletableFuture<SignatureHelp> signatureHelp(TextDocumentPositionParams position) {
+	public CompletableFuture<SignatureHelp> signatureHelp(SignatureHelpParams position) {
 		logInfo(">> document/signatureHelp");
 		SignatureHelpHandler handler = new SignatureHelpHandler(preferenceManager);
 		return computeAsync((monitor) -> handler.signatureHelp(position, monitor));
@@ -573,6 +574,7 @@ public class JDTLanguageServer implements LanguageServer, TextDocumentService, W
 		return computeAsync((monitor) -> handler.findReferences(params, monitor));
 	}
 
+	@Override
 	public CompletableFuture<List<? extends Location>> findLinks(FindLinksParams params) {
 		logInfo(">> java/findLinks");
 		return computeAsync((monitor) -> FindLinksHandler.findLinks(params.type, params.position, monitor));

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java
@@ -46,8 +46,8 @@ import org.eclipse.jdt.ls.core.internal.contentassist.SignatureHelpRequestor;
 import org.eclipse.jdt.ls.core.internal.preferences.PreferenceManager;
 import org.eclipse.lsp4j.SignatureHelp;
 import org.eclipse.lsp4j.SignatureHelpOptions;
+import org.eclipse.lsp4j.SignatureHelpParams;
 import org.eclipse.lsp4j.SignatureInformation;
-import org.eclipse.lsp4j.TextDocumentPositionParams;
 
 public class SignatureHelpHandler {
 
@@ -63,7 +63,7 @@ public class SignatureHelpHandler {
 		this.preferenceManager = preferenceManager;
 	}
 
-	public SignatureHelp signatureHelp(TextDocumentPositionParams position, IProgressMonitor monitor) {
+	public SignatureHelp signatureHelp(SignatureHelpParams position, IProgressMonitor monitor) {
 
 		SignatureHelp help = new SignatureHelp();
 

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandlerTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandlerTest.java
@@ -34,7 +34,7 @@ import org.eclipse.jdt.ls.core.internal.WorkspaceHelper;
 import org.eclipse.jdt.ls.core.internal.preferences.PreferenceManager;
 import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
 import org.eclipse.lsp4j.SignatureHelp;
-import org.eclipse.lsp4j.TextDocumentPositionParams;
+import org.eclipse.lsp4j.SignatureHelpParams;
 import org.junit.Before;
 import org.junit.Test;
 
@@ -407,7 +407,7 @@ public class SignatureHelpHandlerTest extends AbstractCompilationUnitBasedTest {
 
 	private SignatureHelp getSignatureHelp(ICompilationUnit cu, int line, int character) {
 		String payload = createSignatureHelpRequest(cu, line, character);
-		TextDocumentPositionParams position = getParams(payload);
+		SignatureHelpParams position = getParams(payload);
 		return handler.signatureHelp(position, monitor);
 	}
 

